cmd/root.go

```diff
@@ -39,11 +40,24 @@ var (
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
 			log.SetOutput(cmd.OutOrStdout())
+			initConfig()
+
+			if viper.GetString("server") == "" {
+				return fmt.Errorf(style.Red.Render("server is required"))
+			}
+			if viper.GetString("token") == "" {
+				return fmt.Errorf(style.Red.Render("token is required"))
+			}
+
 			client, err := getClient()
 			cobra.CheckErr(err)
 
 			partkeys, err := internal.GetPartKeys(context.Background(), client)
-			cobra.CheckErr(err)
+			if err != nil {
+				return fmt.Errorf(
+					style.Red.Render("failed to get participation keys: %s"),
```

Comment:
<img width="956" alt="Screenshot 2024-11-05 at 20 46 59" src="https://github.com/user-attachments/assets/bfdc1c30-af58-4e03-ada6-36e938384bca">

From when I had it set to localhost 8080 but there was nothing there. In addition to the error itself the help page is printed out. Just wondering if that's intentional, or if we'd prefer to only have the error printed out in red?

Reply:
It's pretty common to show usage whenever it's a configuration error. We could get more granular with the error from the API but it may be better for the TUI daemon in the long run.